libobs/obs-audio.c

```diff
@@ -33,13 +33,49 @@ static void push_audio_tree(obs_source_t *parent, obs_source_t *source, void *p)
 
 	if (da_find(audio->render_order, &source, 0) == DARRAY_INVALID) {
 		obs_source_t *s = obs_source_get_ref(source);
-		if (s)
+		if (s) {
 			da_push_back(audio->render_order, &s);
+			s->audio_is_duplicated = false;
+		}
 	}
 
 	UNUSED_PARAMETER(parent);
 }
 
+static inline bool is_individual_audio_source(obs_source_t *source)
+{
+	return source->info.type == OBS_SOURCE_TYPE_INPUT && (source->info.output_flags & OBS_SOURCE_AUDIO) &&
+	       !(source->info.output_flags & OBS_SOURCE_COMPOSITE);
+}
+
+/* This version of push_audio_tree has the purpose of detecting sources which appear several times in the audio tree.
+ *They are then tagged as such to avoid their mixing in scenes and transitions and mixed directly as root_nodes.
+ */
```

Comment:
```suggestion
/* 
 * This version of push_audio_tree has the purpose of detecting sources which appear several times in the audio tree.
 * They are then tagged as such to avoid their mixing in scenes and transitions and mixed directly as root_nodes.
 */
```

Reply:
done